x/iro/types/bonding_curve.go

```diff
@@ -0,0 +1,80 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	"github.com/osmosis-labs/osmosis/osmomath"
+)
+
+/*
+with the following actions:
+  - SpotPrice(x) = M*x^N + C
+  - Cost(x, x1) = integral(x1) - integral(x)
+    The integral of y = m * x^N + c is (m / (N + 1)) * x^(N + 1) + c * x.
+*/
+
+func NewBondingCurve(m, n, c math.LegacyDec) BondingCurve {
+	return BondingCurve{
+		M: m,
+		N: n,
+		C: c,
+	}
+}
+
+func DefaultBondingCurve() BondingCurve {
+	// linear bonding curve as default
+	return BondingCurve{
+		M: math.LegacyMustNewDecFromStr("0.005"),
+		N: math.LegacyOneDec(),
+		C: math.LegacyZeroDec(),
+	}
+}
+
+// validateBasic checks if the bonding curve is valid
+func (lbc BondingCurve) ValidateBasic() error {
```

Comment:
as discussed with yishay we should probably add some sane valiation here (i.e possible ranges and num of decimals for the power)

Reply:
added 
```
const (
	MaxNValue    = 10
	MaxPrecision = 2
)
```